src/common/log.ts

```diff
@@ -1,15 +1,44 @@
 import * as riverpig from 'riverpig'
+import { LoggerConfig } from 'riverpig'
+import { Logger } from 'riverpig'
+import * as debug from 'debug'
 
 import through2 = require('through2')
 const logStream = through2()
 logStream.pipe(process.stdout)
 
-// TODO: Not clear why I needed this, but got a
-//   TypeScript error without it.
-export interface ConnectorLogger extends riverpig.Logger { }
+export class ConnectorLogger {
+  river: any
+  tracer: any
+
+  constructor (namespace: string, config0?: LoggerConfig) {
+    this.river = riverpig(namespace, config0) 
+    this.tracer = this.river.trace || debug(namespace + ':trace')
+  }
+
+  info (msg: any, ...elements: any[]): void {
+    this.river.info(msg, ...elements)
+  }
+
+  warn (msg: any, ...elements: any[]): void {
+    this.river.warn(msg, ...elements)
+  }
+
+  error (msg: any, ...elements: any[]): void {
+    this.river.error(msg, ...elements)
+  }
+
+  debug (msg: any, ...elements: any[]): void {
+    this.river.debug(msg, ...elements)
+  }
+
+  trace (msg: any, ...elements: any[]): void {
```

Comment:
why require two arguments instead of doing just `(...msg: any[])`?

Reply:
Typescript is enforcing that the function calls here match the function signatures for their interfaces. Both debug and riverpig implement their logging functions like this, so they expect at least one argument and using just the rest parameter will cause it to complain that it got at least 0 args on compile.

Reply:
Should we add a comment about this for others who might have the same question?